Hema/lib/first_project/power.rb

```diff
@@ -0,0 +1,15 @@
+module Hema
+  class Power
+    def self.pow(num, power)
+      result = 1
+      if power >= 0
+        for i in (1..power) do
```

Comment:
What I am gonna say in this comment, I don't know if it is appropiate or if it is too early (to difficult at this stage), so if you find it too hard right now, do not worry and just ignore this comment :joy:

In Ruby we usually write this `for i in (1..power) do` like `(1..power).each do |i|` and what it does is: for the instance `(1..power)` of the [class Range](https://ruby-doc.org/core-2.4.1/Range.html), call its instance method `each` and run the block after the `do` with the `i` as the parameter of the block.

Alternatively, as you do not mind about the index at all, you can also use the instance method `times` of the [class Integer](https://ruby-doc.org/core-2.4.1/Integer.html#method-i-times), which you can use when either you don't care about the value of the index, or when it can start by 0.

So it would be:
```ruby
power.times { result *= num }
```

Considering that `{ ... }` is the same as `do ... end` :smile: .

You canalso use `inject` of an [Enumerable](https://ruby-doc.org/core-2.4.0/Enumerable.html#method-i-inject) (which is mostly any object that can be iterated, like a Range or an Array for example), and then you do not need the `result` variable and it is shorter :smile: Although it is too inneficient for this and I wouldn't recommend it, but since this is for the sake of learning, it would be:
```ruby
if power >= 0
  ([num] * power).inject(1) { |n, result| result * n }
else
  1.0 / pow(num, -power)
end
```
What `([num] * power).inject(1) { |n, result| result * n }` does is, first make an array of `power` elements, all of them containing `num`, and then iterate all of them, with a variable called result starting by `1`, and for each iteration of the array save in that variable the value of `result * num`, which is what returns at the end.

Reply:
And there could be also a recursive version of this method, which in code is usually simpler but you also need to check to what it does with the stack of memory :smile: This version would be:
```ruby
def self.pow(num, power)
  if power == 0
    1
  elsif power > 0
    num * pow(num, power - 1)
  else # power < 0
    1.0 / pow(num, -power)
  end
end
```
And there are probably more efficient versions but for now this is enough :smile: